File: SIMS/CreateSHA.ashx.vb
Imports System.Web
Imports System.Web.Services
Imports System.Data.SqlClient

Public Class CreateSHA
    Implements System.Web.IHttpHandler

    Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest

        Dim site_id As String = context.Request.QueryString("site_id")
        Dim ref_page As String = context.Request.QueryString("ref")

        Using cnx As New SqlConnection(Config.ConnectionInfo)
            cnx.Open()

            Dim sql As String = Nothing
            Dim dt As DataTable = New DataTable()

            sql = "INSERT INTO SHA_Site_Master" & _
                  " (site_id)" & _
                  " VALUES (" & site_id & ")"

            Dim cmd As New SqlCommand(sql, cnx)
            cmd.ExecuteNonQuery()

            cnx.Close()
        End Using

        'If coming from the SHA Status Report page, take them directly to the newly created SHA
        If ref_page = "" Then
            context.Response.Redirect("/SIMSClassic/StationInfo.asp?site_id=" & site_id)
        Else
            context.Response.Redirect("SiteHazardAnalysis.aspx?site_id=" & site_id)
        End If

    End Sub

    ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return False
        End Get
    End Property

End Class

File: Safety/Handler/CreateSHA.ashx.cs
using Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Safety.Helper
{
    /// <summary>
    /// Summary description for CreateSHA
    /// </summary>
    public class CreateSHA : IHttpHandler
    {
        private Data.SIMSDataContext db = new Data.SIMSDataContext();
        public WindowsAuthenticationUser user = new WindowsAuthenticationUser();

        public void ProcessRequest(HttpContext context)
        {
            int site_id = Convert.ToInt32(context.Request.QueryString["site_id"]);

            Data.SHA sha = new Data.SHA() {
                site_id = site_id,
                reviewed_by = user.ID,
                reviewed_dt = DateTime.Now
            };

            db.SHAs.InsertOnSubmit(sha);
            db.SubmitChanges();

            context.Response.Redirect(String.Format("{0}SHAView.aspx?site_id={1}", Config.SafetyURL, site_id));
        }

        public bool IsReusable
        {
            get
            {
                return false;
            }
        }
    }
}